skbio/metadata/_interval.py

```diff
@@ -0,0 +1,756 @@
+                locations.sort()
+                self._locations = locations
+                # reset all the boundaries to True!!
+                del self.boundaries
+
+            # If both `locations` and `boundaries` are provided, set
+            # `self.locations` and `self.boundaries`.
+            else:
+                combo = [i for i in sorted(zip(locations, boundaries))]
+                # use the boundaries setter to validate the value
+                self._boundaries = [i[1] for i in combo]
+                self._locations = [i[0] for i in combo]
+
+            self._interval_metadata._is_stale_tree = True
+
+    @property
+    @experimental(as_of='0.5.0-dev')
+    def boundaries(self):
+        '''The openness of each coordinate.
+
+        This indicates that the exact boundary point of a interval feature
+        is unknown. The location may begin or end at some points outside
+        the specified coordinates. This accommodates the location format [1]_
+        of INSDC.
+
+        References
+        ----------
+        .. [1] ftp://ftp.ebi.ac.uk/pub/databases/embl/doc/FT_current.html#3.4.3
+        '''
+        return self._boundaries
+
+    @boundaries.setter
+    @experimental(as_of='0.5.0-dev')
+    def boundaries(self, value):
+        '''Set ``boundaries``.
+
+        The ``value`` should an iterable matching ``self.locations``.
+        '''
+        self._locations_boundaries_setter(boundaries=value)
+
+    @boundaries.deleter
+    @experimental(as_of='0.5.0-dev')
+    def boundaries(self):
+        '''Delete ``boundaries``.
+
+        This set all boundaries to be True.
+        '''
+        if self.dropped:
+            raise RuntimeError('Cannot change boundaries dropped '
+                               'Interval object.')
+        self._boundaries = [(True, True)] * len(self.locations)
+
+    @property
+    @experimental(as_of='0.5.0-dev')
+    def locations(self):
+        '''The coordinates of the interval feature.
+
+        It should be a list of tuples of int pair. Each tuple stores
+        the start and end coordinates of a span of the interval feature.
+        '''
+        return self._locations
+
+    @locations.setter
+    @experimental(as_of='0.5.0-dev')
+    def locations(self, value):
+        '''Set ``locations``.
+
+        WARNING: setting ``locations`` will reset ``boundaries`` value to True.
+        This is not totally surprising because it is justifiable your old
+        boundaries don't fit the new locations.
+        '''
+        self._locations_boundaries_setter(locations=value)
+
+    @property
+    @experimental(as_of='0.5.0-dev')
+    def metadata(self):
+        '''The metadata of the interval feature.
+
+        It stores the metadata (eg. gene name, function, ID, etc.) of
+        the interval feature as a ``dict``.
+        '''
+        return self._metadata
+
+    @metadata.setter
+    @experimental(as_of='0.5.0-dev')
+    def metadata(self, value):
+        if self.dropped:
```

Comment:
Scratch that, see my comment [here](https://github.com/biocore/scikit-bio/pull/1414/files#r77423594)